src/x/context/context.go

```diff
@@ -316,7 +320,7 @@ func (c *ctx) spanIsSampled(sp opentracing.Span) bool {
 func (c *ctx) StartTraceSpan(name string) (Context, opentracing.Span, bool) {
 	goCtx, exists := c.GoContext()
 	if !exists || c.checkedAndNotSampled {
-		return c, nil, false
+		return c, noopTracer.StartSpan(name), false
```

Comment:
I think you can change the signature to `StartTraceSpan(name string) (Context, opentracing.Span)` to make this easier to use

Reply:
Hey, as per earlier comment I left - knowing whether it's being sampled or not is important so for hot path code we can avoid doing anything that might cause an allocated unless the trace is absolutely being sampled.  Hence the `bool` is important.